components/collection/CollectionList.vue

```diff
@@ -11,20 +11,33 @@
     <LoadPreviousPage
       v-if="startPage > 1 && !isLoading && total > 0"
       @click="reachTopHandler" />
+
     <DynamicGrid
+      v-if="!isLoading && total"
       :id="scrollContainerId"
       grid-size="medium"
-      :default-width="{ small: 16 * 15, medium: 16 * 20, large: 16 * 25 }"
+      :default-width="GRID_DEFAULT_WIDTH"
       :mobile-variant="false">
       <div
         v-for="(collection, index) in collections"
         :key="collection.id"
         :class="scrollItemClassName"
         :data-cy="`collection-index-${index}`">
-        <CollectionCard :is-loading="isLoading" :collection="collection" />
+        <CollectionCard :collection="collection" />
       </div>
     </DynamicGrid>
-    <EmptyResult v-if="total === 0" />
+
+    <DynamicGrid
+      v-else-if="isLoading"
+      :id="scrollContainerId"
+      grid-size="medium"
+      :default-width="GRID_DEFAULT_WIDTH"
+      :mobile-variant="false">
+      <CollectionCard v-for="n in SKELETON_COUNT" :key="n" is-loading />
```

Comment:
@roiLeo updated here to reflect my idea:

I need to wrap the CollectionCard Skeleton in a `<DynamicGrid />` with duplicate props because otherwise the Skeleton Card would not have the right width, which is dynamically calculated by the `<DynamicGrid />` based on the container size and set for the child components.

If the duplicated `<DynamicGrid />` is not an issue, then I have no problem with this approach.

Reply:
With your solution you still kept the logic in `CollectionCard` component (Skeleton & isLoading) and it's working well.
I just feel it's better to have two "different" component but I might be wrong.